File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer = true;

        String[] str = new String[s.length()];
        Stack<Integer> stack = new Stack<>();
        int ans = 0 ;
        for(int i=0; i<s.length(); i++){
            // str[i] = s.substring(i, i+1);
            String tmp = s.substring(i, i+1);
            
            if(tmp.equals("(")){
                ans += 1;
            }
            else{
                if(ans == 0){
                    ans -= 1;
                    answer = false;
                    break;
                }
                else{
                    ans -= 1;
                }
            }
        }
        
        if(ans == 0){
            answer = true;
        }
        else{
            answer = false;
        }
        
        
        // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
        //System.out.println("Hello Java");

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    
    for str in s:
        if str == '(':
            stack.append(str)
            continue
        if len(stack) > 0:
            tmp = stack.pop()
        
            if tmp != '(':
                return False
        else:
            answer = False
    if len(stack) != 0:
        return False
    return answer